### DISCUSSION TITLE
Looks like `through_tables` doesn't work in nested node

### DISCUSSION THREAD

**Discussion by ashrowty** [toluaina/pgsync#301] [General]
if i try to bring in a child table using a `through_table` at the second level, it doesn't work 

structure -

root
    child1
        child2 through_table child1_child2

however i tried the exact same tables at first level and this works -
child1
    child2 through_table child1_child2 

when i look at the stack trace, it doesn't seem to want to go through the `_through()` path. instead it goes through `_non_through()` path - https://github.com/toluaina/pgsync/blob/e4acd7d89899db7abc8fcbb273aac679654d911f/pgsync/querybuilder.py#L739

should i file a bug?

**Comment by toluaina** [comment]
Yes please open an issue with as much details as you can

**Comment by goktugbati** [comment]
I had the same issue and solved it by creating a materialized view.  only difference compared to tables is that you provide primary key and base tables in the schema.json as below

    "nodes": {
      "table": "elastic_form_view",
      "primary_key": [
        "id"
      ],
      "base_tables": [
        "node",
        "space"
      ],
